src/components/PopupWithForm.js

```diff
@@ -15,19 +30,42 @@ export default class PopupWithForm extends Popup {
     });
     return inputValues;
   }
-  // close() {
-  //   this._popupForm.reset();
-  //   super.close();
-  // }
+
+  open() {
+    this._popupForm.reset();
+    super.open();
+  }
 
   setEventListeners() {
     //event listeners
-    this._popupForm.addEventListener("submit", (e) => {
+    this._popup = this._popupForm = this._popupElement.querySelector(".modal__form");
+    this._popup.addEventListener("submit", (e) => {
       e.preventDefault();
       const inputValues = this._getInputValues();
       this._handleFormSubmit(inputValues);
       this.close();
     });
     super.setEventListeners();
   }
+
+  setSubmitAction(action) {
+    this._handleFormSubmit = action;
+  }
+
+  renderLoading(isLoading) {
+    this._modalButton = this._popupElement.querySelector(".modal__button");
```

Comment:
**Could be improved**

Please, find the button, form, inputs only once in the constructor